sample_apps/simple_app_python/example_app.py

```diff
@@ -38,6 +39,7 @@
 
 # ------------------------------------------------------------------------------
 # Script global symbols to app-data dirs.
+CLIENT_APP_DATA='app1_data'
```

Comment:
I think I said this in the previous PR but I will reiterate here -- this naming is confusing and doesn't follow the C++ example. Why do we need two data dirs? The example app should have a single data dir where the state is persisted. The name of this dir should be passed as command line argument (it's ok to have a default name). Then we can start a server instance of this example passing `app1_data` (or whatever) as data dir and another instance passing `app2_data` as data dir.

Having `PROVISIONING_DIR` also doesn't make a lot of sense. Why not simply store the app cert and the root cert in the data dir? The app cert **belongs** to the app and it make sense to be in its data dir.

Reply:
@rgerganov : I think there is some terminology confusion going on here.

Re: '_this naming is confusing and doesn't follow the C++ example._' - Not exactly right. yes, it's true that in the C++ sample program only one data-dir variable is declared. But if you see the invocation from `run_example.sh`, for instance for simple_app, you will see this program is invoked with diff `--data-dir` for client & server.

Here is the output:

```
Running dry-run run_app_by_name_as_server_talk_to_Cert_Service:

/home/agurajada/Code/certifier-framework-for-confidential-computing/sample_apps/simple_app/example_app.exe --data_dir=./app2_data/ --operation=cold-init --measurement_file=example_app.measurement --policy_store_file=policy_store --print_all=true
```

And for the client-app:

```
Running dry-run run_app_by_name_as_client_talk_to_Cert_Service

/home/agurajada/Code/certifier-framework-for-confidential-computing/sample_apps/simple_app/example_app.exe --data_dir=./app1_data/ --operation=cold-init --measurement_file=example_app.measurement --policy_store_file=policy_store --print_all=true
```

So we are doing what you say in your next comment; i.e. _Then we can start a server instance of this example passing app1_data (or whatever) as data dir and another instance passing app2_data as data dir._

---

We need two diff data_dirs, one each for the client & server, because each instance of the app in this client- / server-mode wants to store its trust_data in its own dedicated policy_store file on disk.

That's how the C++ apps work and I have continued the same model for the Python simple_app.

Reply:
About the provisioning dir - this is used by the driver script to generate policy-key and related files in one common dir which are then disbursed to the client- and server's app-data-dirs.

---

I think the Python simple_app is doing everything that the C++ app is also doing.

All you are seeing with these lines of code:

```
 42 CLIENT_APP_DATA='app1_data'
 43 SERVER_APP_DATA='app2_data'
```

is that these symbols are now used in this commit to generate `--help` messages, documenting the ways this program can be invoked. 

See the new messages added from L466 onwards.

Hope these explanations clarify the code-situation and your concerns are addressed.

Reply:
Thanks, somehow I missed that `CLIENT_APP_DATA` and `SERVER_APP_DATA` are used only in the help message. I'd suggest to move them in the `parseargs()` function instead of having them in the global namespace.

I still don't get reason for having `PROVISIONING_DIR` though. Why not store the app cert and the root cert in the app's data dir? Right now you are passing this as `data_dir` argument to `write_certificates_to_file()` and I wonder why not using the data dir specified by the user. Why using hardcoded directory name (`./provisioning`) for this?

Reply:
You are right, @rgerganov -- This is left-over cruft from my dev efforts on this sample program.

Indeed, `run_example.sh` disburses the generated binary cert / key-files from `provisioning` to app-specific data-dirs. So, yes, you are right; I should no longer need to rely on provisioning dir.

Cleaned up the program to address both your comments.

Thanks, for a careful review on this point ... and also for spotting the typo in `write_certficates_to_file()`.